[Docs] Add 1.27 generative-friendliai docs (#1)
friendliai/weaviate-io: developers/weaviate/model-providers/friendliai/generative.md

---
title: Generative AI
sidebar_position: 51
image: og/docs/integrations/provider_integrations_friendliai.jpg
# tags: ['model providers', 'friendliai', 'generative', 'rag']
---

# FriendliAI Generative AI with Weaviate

import BetaPageNote from '../_includes/beta_pages.md';

<BetaPageNote />

:::info Added in `v1.27.0`
:::

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import FilteredTextBlock from '@site/src/components/Documentation/FilteredTextBlock';
import PyConnect from '!!raw-loader!../_includes/provider.connect.py';
import TSConnect from '!!raw-loader!../_includes/provider.connect.ts';
import PyCode from '!!raw-loader!../_includes/provider.generative.py';
import TSCode from '!!raw-loader!../_includes/provider.generative.ts';

> Weaviate's integration with FriendliAI's APIs allows you to access open source and their models' capabilities directly from Weaviate.

Reviewer: I don't think this is a correct sentence (access open source models?).

Suggestion:
Weaviate's integrations with FriendliAI's APIs allow you to access open-source models on Friendli and utilize models' capabilities directly from Weaviate.

> Weaviate's integrations with FriendliAI's APIs allow you to access open-source models on Friendli and utilize models' capabilities directly from Weaviate.

Reviewer 2: ```suggestion
Weaviate's integrations with FriendliAI's APIs allow you to access open-source models on Friendli Suite and utilize models' capabilities directly from Weaviate.
```

> [Configure a Weaviate collection](#configure-collection) to use an FriendliAI generative AI model, and Weaviate will perform retrieval augmented generation (RAG) using the specified model and your FriendliAI API key.

Reviewer: ```suggestion
[Configure a Weaviate collection](#configure-collection) to use generative AI models on FriendliAI. Weaviate will perform Retrieval Augmented Generation (RAG) using the specified model and your FriendliAI API key.
```

> More specifically, Weaviate will perform a search, retrieve the most relevant objects, and then pass them to the FriendliAI generative model to generate outputs.

Reviewer: ```suggestion
More specifically, Weaviate will perform a search, retrieve the most relevant objects, and then pass them to the FriendliAI generative AI model to generate outputs.
```

![RAG integration illustration](../_includes/integration_friendliai_rag.png)

## Requirements

### Weaviate configuration

> Your Weaviate instance must be configured with the FriendliAI generative AI integration (`generative-friendliai`) module.

Reviewer 2: ```suggestion
Your Weaviate instance must be configured with the FriendliAI generative AI integration (`generative-friendli`) module.
```

Author: I believe the module name itself should be the full name of our company.
Compared to API key or Baseurl, the module name is more like a standalone form than a descriptive form.

<details>

>   <summary>For Weaviate Cloud (WCD) users</summary>

Reviewer: Random question but why is Weaviate Cloud WC**D**?

Reviewer 2: Also interested

This integration is enabled by default on Weaviate Cloud (WCD) serverless instances.

</details>

<details>
  <summary>For self-hosted users</summary>

- Check the [cluster metadata](../../config-refs/meta.md) to verify if the module is enabled.
- Follow the [how-to configure modules](../../configuration/modules.md) guide to enable the module in Weaviate.

</details>

### API credentials

> You must provide a valid FriendliAI API key to Weaviate for this integration. Go to [FriendliAI](https://docs.friendli.ai/openapi/create-chat-completions) to sign up and obtain an API key.

Reviewer 3: Is it valid link?

Author: Yes, It is

Reviewer 3: But description says "go to ... sign up and obtain an API key". Isn't it should be `suite.friendi.ai`?

Reviewer 3: @snowman39

> You must provide a valid FriendliAI API key to Weaviate for this integration. Go to [FriendliAI](https://docs.friendli.ai/openapi/create-chat-completions) to sign up and obtain an API key.

Reviewer 2: ```suggestion
You must provide a valid Friendli Suite API key to Weaviate for this integration. Go to [Friendli Suite](https://docs.friendli.ai/openapi/create-chat-completions) to sign up and obtain an API key.
```

      startMarker="# START FullGenerativeFriendliAI"
      endMarker="# END FullGenerativeFriendliAI"
      language="py"
    />
  </TabItem>

  <TabItem value="js" label="JS/TS API v3">
    <FilteredTextBlock
      text={TSCode}
      startMarker="// START FullGenerativeFriendliAI"
      endMarker="// END FullGenerativeFriendliAI"
      language="ts"
    />
  </TabItem>

</Tabs>

For further details on model parameters, see the [FriendliAI API documentation](https://docs.friendli.ai/openapi/create-chat-completions).

### Available models

* `meta-llama-3-8b-instruct`
* `meta-llama-3-70b-instruct`
* `mixtral-8x7b-instruct-v0-1`

## Further resources

### Code examples

Once the integrations are configured at the collection, the data management and search operations in Weaviate work identically to any other collection. See the following model-agnostic examples:

- The [how-to: manage data](../../manage-data/index.md) guides show how to perform data operations (i.e. create, update, delete).
- The [how-to: search](../../search/index.md) guides show how to perform search operations (i.e. vector, keyword, hybrid) as well as retrieval augmented generation.

### References

> - FriendliAI [API documentation](https://docs.friendli.ai/openapi/create-chat-completions)

Reviewer 3: p9)
In line 232, link is set in `[FriendliAI API documentation]`.
But in line 251, link is set in `FriendliAI [API documentation]`.

Author: Thanks for the finding 🙏
